### ISSUE TITLE
dotnet vstest only supports one framework at a time with multiple test projects

### ISSUE THREAD

**Issue by gtbuchanan** [microsoft/vstest#1768]
## Steps to reproduce
Running the following command fails in our environment because we have multiple test assemblies targeting different frameworks (netcoreapp2.0 & net461):

```
dotnet vstest --Parallel --Platform:x64 --logger:trx (Get-ChildItem -Recurse src\**\bin\**\Release\**\*.Test.dll)
```

## Expected behavior
dotnet vstest should infer the framework for each assembly and create a single output file

## Actual behavior
The test run is aborted with "Unable to find tests" because all test assemblies are assumed to use the same framework. I have to run the command once for each target framework instead, which produces multiple `trx` files.

```
dotnet vstest --Parallel --Platform:x64 --logger:trx (Get-ChildItem -Recurse src\**\bin\**\Release\netcoreapp2.0\*.Test.dll)
dotnet vstest --Parallel --Platform:x64 --Framework:.NETFramework,Version=v4.6.1 --logger:trx (Get-ChildItem -Recurse src\**\bin\**\Release\net461\*.Test.dll)
```

## Diagnostic logs
![image](https://user-images.githubusercontent.com/715687/45859106-3b885f00-bd26-11e8-9888-8122076cfe58.png)

## Environment
Windows 10 Enterprise 1803
vstest.console 15.8.0

## Additional Info
Moved from [VS Developer Community](https://developercommunity.visualstudio.com/content/problem/334358/dotnet-vstest-only-supports-one-framework-at-a-tim.html)
Related to #705

[AB#1408158](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1408158)

**Comment by smadala** [comment]
Currently `dotnet vstest` supports only running one kind of target frameworks(Full or Core). This is as per design. 

We suggest running `dotnet vstest` for each target framework separately.

**Comment by denxorz** [comment]
We also experienced a simular problem when running tests on TFS. Our solution contains 10 .NET Core tests dlls and one .NET Full dll. These tests are picked up by the default filter:
```
**\*test*.dll
!**\*TestAdapter.dll
!**\obj\**
```
The test session is not directly aborted when the .NET Full is encountered, but it seems to be later. At the end of the test run it only specifies "Test Run Aborted" without any reason. It would have really helped us if it at least specified why the tests where aborted.

Specifying all the tests seperatly does not seems to be a really maintainable option. We will try to look into the new `dotnet test` command.

**Comment by nohwnd** [comment]
The ability to run multiple TFMs and multiple architectures at the same time will be shipped in 17.3 (there are already previews available). You should avoid using the platform and framework options, because they force a single unified platform/framework on all the dlls, instead of letting autodetection figure out how is each dll built.
